### PR TITLE
Add option converterClass to @ParamField

### PR DESCRIPTION
**Pull request by mhagnumdw**
Continuation of the PR #417 

@decebals Please tell me about any changes to make.

### CODE REVIEW

File: pippo-core/src/main/java/ro/pippo/core/ParamField.java [hunk 1]
```diff
@@ -33,4 +35,6 @@
 
     String pattern() default "";
 
+    Class<? extends Converter> converterClass() default Converter.class;
```
[the remark this answers is not in the document]
Comment (mhagnumdw) [reply pippo-core/src/main/java/ro/pippo/core/ParamField.java]: It is possible, however the options are:
- Class<?> converterClass() default void.class;
- Class converterClass() default void.class;

[the remark this answers is not in the document]
Comment (decebals) [reply pippo-core/src/main/java/ro/pippo/core/ParamField.java]: I think `Class converterClass() default void.class; `

File: pippo-core/src/main/java/ro/pippo/core/Request.java [hunk 2]
```diff
@@ -254,38 +256,43 @@ public void internalSetPathParameters(Map<String, String> pathParameters) {
...
+                                value = getParameters().get(parameterName).toCollection(collectionClass, genericClass, pattern);
+                            }
                         } else {
-                            Class<? extends Collection> collectionClass = (Class<? extends Collection>) fieldClass;
-                            value = getParameters().get(parameterName).toCollection(collectionClass, genericClass, pattern);
+                            value = getParameters().get(parameterName).to(fieldClass, pattern);
                         }
                     } else {
-                        value = getParameters().get(parameterName).to(fieldClass, pattern);
+                        value = getParameters().get(parameterName).convert((Class<? extends Converter<T>>) convertClass, pattern);
```
[the remark this answers is not in the document]
Comment (mhagnumdw) [reply pippo-core/src/main/java/ro/pippo/core/Request.java]: I agree. We can see this soon enough.

File: pippo-core/src/main/java/ro/pippo/core/ParameterValue.java [hunk 3]
```diff
@@ -420,6 +421,15 @@ public Timestamp toSqlTimestamp(Timestamp defaultValue) {
         }
     }
 
+    public <T, C extends Converter<T>> T convert(Class<C> converterClass, String pattern) {
```
Comment (decebals) [inline pippo-core/src/main/java/ro/pippo/core/ParameterValue.java]: Do we need this (static) method in this class? We can move the content of this method directly in Request class where we call ParameterValue#convert.
What you say?

Comment (mhagnumdw) [reply pippo-core/src/main/java/ro/pippo/core/ParameterValue.java]: I think I did not understand.

Comment (decebals) [reply pippo-core/src/main/java/ro/pippo/core/ParameterValue.java]: I reviewed my idea and I think that is not add value, so I gave it up.

**Comment by decebals** [comment]
Thanks!

**Comment by mhagnumdw** [comment]
Thanks too!

What do you think of starting an issue to discuss a refactory in https://github.com/pippo-java/pippo/blob/937aa1b968ea53e7e19f4058aad605c5595428c3/pippo-core/src/main/java/ro/pippo/core/Request.java#L245
?

**Comment by decebals** [comment]
@mhagnumdw 

> What do you think of starting an issue to discuss a refactory in ...

As I said in a previous review's comment in this PR, I think that this methods is relative big and we can move it in other class (PippoUtils, RequestUtils or a better variant/location) as a static method, to reduce the number of lines of `Request` class. We need to pass as parameter in the new (static) method, the parameters map or better the whole request. Maybe the same approach/treatment for `createEntityFromBody` method.  To preserve the backward compatibility as API, we can preserve the signature of this method(s) in `Request` and forward to the new method(s).

What you think?

**Comment by mhagnumdw** [comment]
I agree.

**Pull request merged**